WcaOnRails/app/models/competition_event.rb

```diff
@@ -21,4 +21,31 @@ def to_wcif
       "rounds" => self.rounds.map(&:to_wcif),
     }
   end
+
+  def load_wcif!(wcif)
+    # Remove extra rounds.
+    self.rounds.each do |round|
+      wcif_round = wcif["rounds"][round.number - 1]
```

Comment:
Interesting options! 1) and 2) are changes to the WCIF spec, which is a nice place to fix this, as it affects everyone who uses the WCIF. On the other hand, 3) and 4) are things that people dealing with WCIF data should probably do, but we cannot support libraries for all the possible languages that people will be dealing with WCIF data in.

So, I'd rather address this with a change to the WCIF spec if possible. Here are all the possibilities I can think of:

1. Add a dummy "zeroth" round to the arrays. This is your 1) from above.
2. Use key-value semantics rather than arrays. This is your 2) from above.
3. Instead of giving rounds `number`-s, give them `index`-s (that are indexed from 0).
4. Do nothing.

I think right now I'm leaning towards 3) or 4). What do you think?